test/jdk/javax/rmi/ssl/SSLSocketParametersTest.java

```diff
@@ -116,154 +130,98 @@ public ServerSocket createServerSocket(int port) throws IOException {
-                client.runClient(stub);
-                System.out.println(msg2);
-            } catch (Exception e) {
-                System.out.println(msg3 + " Exception: " + e.toString());
-                e.printStackTrace(System.out);
-                System.exit(1);
-            }
-            break;
-        case 2: /* non-default constructor - default config */
-            System.out.println(msg1);
-            try {
-                HelloImpl server = new HelloImpl(
-                          0,
-                          new ClientFactory(),
-                          new ServerFactory(null,
-                                            null,
-                                            false));
-                Remote stub = server.runServer();
-                HelloClient client = new HelloClient();
-                client.runClient(stub);
-                System.out.println(msg2);
-            } catch (Exception e) {
-                System.out.println(msg3 + " Exception: " + e.toString());
-                e.printStackTrace(System.out);
-                System.exit(1);
-            }
-            break;
-        case 3: /* needClientAuth=true */
-            System.out.println(msg1);
-            try {
-                HelloImpl server = new HelloImpl(
-                          0,
-                          new ClientFactory(),
-                          new ServerFactory(null,
-                                            null,
-                                            null,
-                                            true));
-                Remote stub = server.runServer();
-                HelloClient client = new HelloClient();
-                client.runClient(stub);
-                System.out.println(msg2);
-            } catch (Exception e) {
-                System.out.println(msg3 + " Exception: " + e.toString());
-                e.printStackTrace(System.out);
-                System.exit(1);
+            /* default constructor - default config */
+            case 1 -> run(new ServerFactory(), false);
+
+            /* non-default constructor - default config */
+            case 2 -> run(new ServerFactory(null, null, false), false);
+
+            /* needClientAuth=true */
+            case 3 -> run(new ServerFactory(null, null, null, true), false);
+
+            /* server side dummy_ciphersuite */
+            case 4 -> {
+                try {
+                    run(new ServerFactory(SSLContext.getDefault(),
+                            new String[]{"dummy_ciphersuite"}, null, false),
+                            true);
+                } catch (NoSuchAlgorithmException exc) {
+                    throw new RuntimeException("Could not create SSLContext.", exc);
+                } catch (IllegalArgumentException exc) {
+                    if (!exc.getMessage().toLowerCase().contains("unsupported ciphersuite")) {
+                        throw exc;
+                    }
+                }
```

Comment:
The expected result of cases 4 and 5 is that an IllegalArgumentException is thrown because of the unsupported ciphersuite. The original code just caught `Exception` which would hide a problem in the test e.g., if `SSLContext.getDefault()` throws the NoSuchAlgorithmException.

Reply:
OK - but the run( ) method catches IllegalArgumentException already?

Reply:
The exception is actually being thrown from the `ServerFactory` constructor.  I updated the code to remove the call to `run()` and added a comment clarifying the exception handling.